src/tlo/methods/contraception.py

```diff
@@ -645,24 +645,89 @@ def select_contraceptive_following_birth(self, mother_id, mother_age):
+        # # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
+        # _cons_codes['IUD'] = get_items_from_pkg('IUD')
+        # _cons_codes['injections'] = get_items_from_pkg('Injectable')
+        # _cons_codes['implant'] = get_items_from_pkg('Implant')
+        # _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        # assert set(_cons_codes.keys()) == set(self.states_that_may_require_HSI_to_switch_to)
+        # # items used when initiating a modern reliable method after not using or switching from non-reliable method
+        # _cons_codes['co_initiation'] = get_items_from_pkg('Contraception initiation')
+
+        # ### Get item codes from item names and define number of units per case here
+        get_item_code = self.sim.modules['HealthSystem'].get_item_code_from_item_name
 
         _cons_codes = dict()
-        # items for each method that requires an HSI to switch to
-        _cons_codes['pill'] = get_items_from_pkg('Pill')
-        _cons_codes['male_condom'] = get_items_from_pkg('Male condom')
-        _cons_codes['other_modern'] = get_items_from_pkg('Female Condom')
-        # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
-        _cons_codes['IUD'] = get_items_from_pkg('IUD')
-        _cons_codes['injections'] = get_items_from_pkg('Injectable')
-        _cons_codes['implant'] = get_items_from_pkg('Implant')
-        _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        # # items for each method that requires an HSI to switch to
+        # TODO: @TimH Pills are alternatives, we weight their use by the probabilities (as approximation of use in 80:20
+        #  cases, for each case 0.8 and 0.2 proportion of units is used). Is that okay, or do we want to choose for each
+        #  case only one type of pills? Total usage should be approximately same.
+        # TODO: @Sakshi what are the numbers of pills in monthly packets for both following items?
+        _cons_codes['pill'] =\
+            {get_item_code("Levonorgestrel 0.0375 mg, cycle"): 28*3.75*0.2,  # progesterone-only pills used in 20% cases
+             get_item_code("Levonorgestrel 0.15 mg + Ethinyl estradiol 30 mcg (Microgynon), cycle"): 21*3.75*0.8
+             # combined pills used in other 80% cases
+             }
+        _cons_codes['male_condom'] =\
+            {get_item_code("Condom, male"): 30}
+        _cons_codes['other_modern'] =\
+            {get_item_code("Female Condom_Each_CMST"): 30}
+        _cons_codes['IUD'] =\
+            {get_item_code("Glove disposable powdered latex medium_100_CMST"): 2,
+             get_item_code("IUD, Copper T-380A"): 1}
+        # TODO: originally used "Medroxyprogesterone acetate injection 150mg/mL, 1mL vial with 2ml syringe with 22g 0.7
+        #  X 25mm needle_each_CMST", which is not within the RF_Costing anymore, does it mean we need to add a syringe
```

Comment:
These look very similar to me. In principle, you can use items from the original consumables resource file (don't have to limit to the costing file, right Sakshi?)

Reply:
@sakshimohan, what is better for you if I use? The one which is there now--Medroxyprogesterone Acetate only--, or the one including syringe and needle? If the former, do I need to add a syringe and a needle or is it covered within the cost of "Depot-Medroxyprogesterone Acetate 150 mg - 3 monthly" even if not included in the name? The "Medroxyprogesterone acetate injection 150mg/mL, 1mL vial with 2ml syringe with 22g 0.7 X 25mm needle_Each_BB049500_CMST" is in the `RF_Costing/consumables` in the column `Item_EHP` in the line with item "Depot-Medroxyprogesterone Acetate 150 mg - 3 monthly".

Reply:
Hi @EvaJanouskova. I've current got the cost of this item `Medroxyprogesterone acetate injection 150mg/mL, 1mL vial with 2ml syringe with 22g 0.7 X 25mm needle_Each_BB049500_CMST` from CMST which already includes the syringe and injection. This is for `item_code` = 3 in the model. So perhaps easiest to include the one including syringe and needle?

Reply:
Sorry if I'm missing something here Eva. The name of the consumable for item_code 3 in ResourceFile_Items_and_Packages  is "Medroxyprogesterone acetate injection 150mg/mL, 1mL vial with 2ml syringe with 22g 0.7 X 25mm needle_Each_BB049500_CMST, ampule". For item_code 3, there is availability data in in ResourceFile_Consumables_availability_small and there is cost data in ResourceFile_Costing.
The cost for this item is based on the item with the same name in the CMST costing catalogue which includes the cost of the syringe and needle.
I'm not sure why the name of the consumable in ResourceFile_Costing appears to be Depot-Medroxyprogesterone Acetate 150 mg - 3 monthly but in either case, it's fine for you to go ahead and use item_code 3 for your purposes. If in your scrip for the contraception.py module, you're calling teh consumable by its name, this will be obtained from  ResourceFile_Items_and_Packages so you can ignore the name in ResourceFile_Costing .

Reply:
So the preference is for you to use item code 3. However, if for any other reason, you wish to use item code 1839, I will need to make a few changes to make sure that there is availability data for 1839 in `ResourceFile_Consumables_availability_small` and add cost data to `ResourceFile_Costing`.